micrometer-core/src/main/java/io/micrometer/core/instrument/binder/logging/Log4j2Metrics.java

```diff
@@ -79,11 +84,27 @@ public Log4j2Metrics(Iterable<Tag> tags, LoggerContext loggerContext) {
     @Override
     public void bindTo(MeterRegistry registry) {
         Configuration configuration = loggerContext.getConfiguration();
+        registerMetricsFilter(configuration, registry);
+        loggerContext.updateLoggers(configuration);
+
+        PropertyChangeListener changeListener = listener -> {
+            if (listener.getNewValue() instanceof Configuration && listener.getOldValue() != listener.getNewValue()) {
+                registerMetricsFilter((Configuration) listener.getNewValue(), registry);
```

Comment:
I guess after doing this, we should call `updateLoggers` but that's another change event so we have an infinite loop. Can we break that loop another way and still call `updateLoggers` in `registerMetricsFilter` and in `removeMetricsFilter`?

Reply:
It's actually no problem as we only react on changes where `listener.getOldValue() != listener.getNewValue()`, so triggering `updateLoggers` with the same configuration will not register a new metrics filter (as it should already be registered when its not a new `Configuration` instance).

So I'll just add a call to `updateLoggers`!